colvartools/plot_colvars_traj.py

```diff
@@ -214,11 +215,32 @@ def read_files(self, filenames, list_variables=False,
                 if (step == last_step): continue
                 if ((self._frame >= first) and (self._frame <= last) and
                     (self._frame % every == 0)):
-                    self._parse_line(line)
+                    self._parse_line(line, dict_buffer)
                 self._frame += 1
                 last_step = step
+            for key in dict_buffer:
+                if (key not in self._colvars):
+                    self._colvars[key] = Colvar_traj(key)
+                    self._colvars[key]._set_num_dimensions(dict_buffer[key]['dimension'])
+                cv = self._colvars[key]
+                cv._step = np.concatenate((cv.steps, dict_buffer[key]['cv_step']), axis=0)
+                cv._colvar = np.concatenate((cv.values, dict_buffer[key]['cv_values']), axis=0)
             f.close()
 
+    def as_pandas(self, keys=None):
+        """
+        Export data to pandas DataFrame
+        keys : list of strings
+            list of keys to be read
+        """
+        import pandas as pd
+        tmp_dict = dict()
+        if keys is None:
+            keys = self.variables
+        for key in keys:
+            tmp_dict[key] = self.__getitem__(key).values.tolist()
```

Comment:
2D numpy array doesn't work. You can try the following code:
```python
import numpy as np
import pandas as pd
X = [1,2]
Y = [[1,3],[2,4]]
pd.DataFrame({'X': X, 'Y': Y})
pd.DataFrame({'X': np.array(X), 'Y': np.array(Y)})
```

Reply:
On the other hand, this suggestion would break up the columns, invalidating the ability to access a vector variable (as a Python list) through a single key in Pandas. I looked into passing structured arrays, but unfortunately Pandas requires calling `numpy.isnan()` on the input, which I'm not sure how to circumvent.

So it boils down to what people's preferences are for data layout in Pandas. I'm personally fine with anything, including what you suggested. If the Pandas conversion takes too long to be done interactively, people are still able to append `.as_pandas()` immediately upon constructing. Leave this as-is.